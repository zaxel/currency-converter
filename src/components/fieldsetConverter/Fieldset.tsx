import React from 'react';
import Form from 'react-bootstrap/Form';
import styles from './styles.module.css';
import { IFieldsetProps } from '../../interfaces';



const Fieldset = ({onInputChange, legend, options=[], val, isDisabled}: IFieldsetProps) => {
    return (
        <fieldset>
            <legend>{legend}</legend>

            <Form.Control disabled={isDisabled} value={+val} onChange={onInputChange} className={styles.input_number} type="number" placeholder="amount" step={0.1} />

            <Form.Select className={styles.select}>
                {options.map((op,i)=>{
                    if(i===0)
                        return <option>{op}</option>
                        return <option value={i}>{op}</option> 
                })}
                
            </Form.Select>
        </fieldset>
    );
};

export default Fieldset;