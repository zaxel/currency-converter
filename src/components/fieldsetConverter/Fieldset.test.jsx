// Fieldset.test.js
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Fieldset from './Fieldset';

describe('Fieldset component', () => {
    test('renders Fieldset component with legend and inputs', () => {
        // Arrange
        const props = {
            onInputChange: jest.fn(),
            val: 42,
            onSelectChange: jest.fn(),
            selectVal: 'Option 1',
            legend: 'Test Legend',
            options: ['Option 1', 'Option 2'],
            isDisabled: false,
        };

        // Act
        render(<Fieldset {...props} />);

        // Assert
        expect(screen.getByText('Test Legend')).toBeInTheDocument();
        expect(screen.getByPlaceholderText('amount')).toHaveValue(42);
        expect(screen.getByDisplayValue('Option 1')).toBeInTheDocument();
    });

    test('calls onInputChange and onSelectChange when inputs change', () => {
        // Arrange
        const onInputChange = jest.fn();
        const onSelectChange = jest.fn();
        const props = {
            onInputChange,
            val: 42,
            onSelectChange,
            selectVal: 'Option 1',
            legend: 'Test Legend',
            options: ['Option 1', 'Option 2'],
            isDisabled: false,
        };

        //Act
        render(<Fieldset {...props} />);
        fireEvent.change(screen.getByPlaceholderText('amount'), { target: { value: 50 } });
        fireEvent.change(screen.getByRole('combobox'), { target: { value: 'Option 2' } });
        // Assert
        expect(onInputChange).toHaveBeenCalledWith(expect.anything());
        expect(onSelectChange).toHaveBeenCalledWith(expect.anything());
    });

    test('snapshot test', () => {
        // Arrange
        const onInputChange = jest.fn();
        const onSelectChange = jest.fn();
        const props = {
            onInputChange,
            val: 42,
            onSelectChange,
            selectVal: 'Option 1',
            legend: 'Test Legend',
            options: ['Option 1', 'Option 2'],
            isDisabled: false,
        };
        
        //Act
        const view = render(<Fieldset {...props} />);
        
        // Assert
        expect(view).toMatchSnapshot();
    })
})

// Add more tests based on your specific requirements
