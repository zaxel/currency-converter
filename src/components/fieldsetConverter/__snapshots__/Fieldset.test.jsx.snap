// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fieldset component snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <fieldset>
        <legend>
          Test Legend
        </legend>
        <input
          class="input_number form-control"
          placeholder="amount"
          step="0.1"
          type="number"
          value="42"
        />
        <select
          class="select form-select"
        >
          <option
            value="Option 1"
          >
            Option 1
          </option>
          <option
            value="Option 2"
          >
            Option 2
          </option>
        </select>
      </fieldset>
    </div>
  </body>,
  "container": <div>
    <fieldset>
      <legend>
        Test Legend
      </legend>
      <input
        class="input_number form-control"
        placeholder="amount"
        step="0.1"
        type="number"
        value="42"
      />
      <select
        class="select form-select"
      >
        <option
          value="Option 1"
        >
          Option 1
        </option>
        <option
          value="Option 2"
        >
          Option 2
        </option>
      </select>
    </fieldset>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
